Adicionando política de pull requests (#74)
fga-eps-mds/2022-2-Alectrion-DOC: docs/Documentos/politicas-branch.md

Outras branches devem ser criadas a partir do padrão: tipo/artefato.

## 4. Pull Requests

### *Labels*

Neste projeto utilizaremos algumas labels para associar às _Issues_ e às _Pull Requests_ para auxiliar no entendimento do objetivo das mesmas. As labels que serão usadas serão listadas a seguir mas também podem ser acessadas [aqui](https://github.com/fga-eps-mds/2022-2-Alectrion-DOC/labels).

 - **ANALYTICS:** Label para associar a questões de métricas e análises;
 - **ARQ:**
 - **bug:** Label para associar a algo que não está funcionando da forma esperada;
 - **DEVOPS:** Label para associar a _issues_ que envolvem os operadores do sistema;
 - **DOCS:** Label para associar a questões de documentação do produto;
 - **duplicate:** Label para marcar _issues_ duplicadas;
 - **EASY:** Label para associar a _issues_ de fácil resolução;

>  - **enhancement:** Label para associar a _issues_ de novas _features_;

Reviewer: Aqui seria issues de melhoria, certo?
```suggestion
 - **enhancement:** Label para associar a _issues_ de melhorias;
```

Reviewer 2: De acordo

Author: De acordo com a descrição da label [aqui](https://github.com/fga-eps-mds/2022-2-Alectrion-DOC/labels), diz que é uma nova feature, mas se acharem que deve ser trocado para melhoria mesmo, posso trocar.

 - **Epic:** Label para associar a épicos;
 - **EPS:** Label para associar a _issues_ que devem ser realizadas ou monitoradas por alunos da disciplina de EPS;
 - **FEATURE:** Label para associar a _features_;
 - **good first issue:** Label para associar a _issues_ que podem ser resolvidas por recém chegados no repositório;
 - **HARD:** Label para associar a _issues_ que possuem maior complexidade de resolução;
 - **help wanted:** Label para associar a _issues_ onde se precisa de ajuda para continuar sua resolução;
 - **HOTFIX:** Label para associar a _issues_ que precisam resolver uma situação muito específica;
 - **invalid:** Label para associar a _issues_ onde algo não parece estar funcionando corretamente;

>  - **MDS:** Label para associar a _issues_ que devem/podem ser realizadas por alunos da disciplina de EPS;

Reviewer: ```suggestion
 - **MDS:** Label para associar a _issues_ que devem/podem ser realizadas por alunos da disciplina de MDS;
```

Reviewer 2: De acordo

 - **MEDIUM:** Label para associar a _issues_ que possuem certa complexidade;
 - **question:** Label para associar a _issues_ onde maiores informações são necessárias para continuar sua resolução;
 - **US:** Label para associar a histórias de usuário;
 - **wontfix:** Label para associar a _issues_ que não possuem intenção de serem solucionadas a princípio;

### Abertura e conteúdo de *Pull Requests*

> Após ter resolvido uma issue e estiver satisfeito com o que foi alterado nos arquivos locais, deve-se enviar essas alterações para o repositório no GitHub por meio de commits. Após isso, pode-se ir para o repositório e abrir um _pull request_, onde deve-se escolher a _branch_ que se quer propor as alterações e informar o máximo de detalhes da contribuição, criando então um _pull request_ para ser revisado.

Reviewer: ```suggestion
Após ter resolvido uma issue e estar satisfeito com o que foi alterado nos arquivos locais, deve-se enviar essas alterações para o repositório no GitHub por meio de commits. Após isso, pode-se ir para o repositório e abrir um _pull request_, onde deve-se escolher a _branch_ que se quer propor as alterações e informar o máximo de detalhes da contribuição, seguindo o [template de _pull requests_](https://github.com/fga-eps-mds/2022-2-Alectrion-DOC/blob/main/.github/pull_request_template.md), criando então um _pull request_ para ser revisado.
```

Reviewer 2: De acordo

### Revisores

> Ao abrir o _pull request_ deve-se marcar ao menos dois integrantes de EPS para revisar as alterações sugeridas para, caso necessário, ser solicitado a fazer edições ou melhorias.

Reviewer 2: ```suggestion
Ao abrir o _pull request_ deve-se marcar ao menos dois integrantes de EPS e dois de MDS para revisarem o conteúdo do PR. Caso seja necessário, o autor do PR será solicitado pelos revisores para fazer edições ou melhorias.
```

> Sempre que implementar alterações sugeridas, deve-se avisar aos revisores para que eles dêem uma nova olhada. Ao fim, se a contribuição for aceita, o _pull request_ será _mergeado_ a branch principal do repositório.

Reviewer: ```suggestion
Sempre que implementar alterações sugeridas, deve-se avisar aos revisores para que eles dêem uma nova olhada. Ao fim, se a contribuição for aceita, o _pull request_ será _mergeado_ à branch de destino do repositório.
```

Reviewer 2: De acordo

## 5. Referências

> [1] EQUIPE ALECTRION 2022-1. Política de Branches. Disponível em: https://fga-eps-mds.github.io/2022-1-Alectrion-DOC/documentation/Documentos/politicas-branch.html

> [2] ATLASSIAN. Gitflow Workflow. Disponível em: https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow

## 6. Histórico de versão

|**Data**|**Descrição**|**Autore(es)**|
|--------|-------------|--------------|
|09/12/2022| Criação da política de _branches_ | Erick Giffoni |

> |16/12/2022| Adição da política de _pull requests_ | Álvaro Leles |

Reviewer 2: ```suggestion
|16/12/2022| Adição da política de _pull requests_ | Álvaro Leles |
|16/12/2022| Revisão da política de _pull requests_ | Aline Helena, Erick Giffoni |
```
